module/zfs/vdev_label.c

```diff
@@ -508,6 +508,11 @@ vdev_config_generate(spa_t *spa, vdev_t *vd, boolean_t getstats,
 		}
 	}
 
+	if (flags & VDEV_CONFIG_L2CACHE)
+		/* indicate that we support L2ARC persistency */
+		VERIFY(nvlist_add_boolean_value(nv,
+		    ZPOOL_CONFIG_L2CACHE_PERSISTENT, B_TRUE) == 0);
```

Comment:
l2arc devices have special labels written to them, see `vdev_label_init` after line 1076, so this won't be added unless you move it in to that block.  We've talked about giving l2arc devices full labels in the past.  This would be nice since it would allow tools like `blkid` to correctly identify them.  This PR seems like the right time to make that change.

```
zdb -l nvme4n1
------------------------------------
LABEL 0
------------------------------------
    version: 5000
    state: 4
    guid: 16243264221632350512
    labels = 0 1 2 3 
```

Reply:
Agreed, but we also need this block at its current position. There is a check at line 1900  in `module/zfs/spa.c`. Otherwise each time `vdev_config_generate()` is called the new value is not added, and L2ARC rebuild is not run.